### PR TITLE
[BEAM-7660] Create Python ParDo load test job on Flink

### PR DESCRIPTION
**Pull request by kamilwu**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Python Load Tests ParDo Flink Batch

**Comment by kamilwu** [comment]
@iemejia  Could you take a look?

**Comment by kamilwu** [comment]
R: @iemejia Could you take a look?

**Comment by kamilwu** [comment]
@iemejia Please take a look. You can also tag someone else if you're busy

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Python Load Tests ParDo Flink Batch

**Comment by kamilwu** [comment]
R: @lgajowy

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Python Load Tests ParDo Flink Batch

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import Docker
+
+String now = new Date().format("MMddHHmmss", TimeZone.getTimeZone('UTC'))
+
+def scenarios = { datasetName, sdkHarnessImageTag -> [
+        [
+                title        : 'ParDo Python Load test: 2GB 100 byte records 10 times',
+                itClass      : 'apache_beam.testing.load_tests.pardo_test:ParDoTest.testParDo',
+                runner       : CommonTestProperties.Runner.PORTABLE,
+                jobProperties: [
```
Comment (lgajowy) [inline .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: that could be named `pipelineOptions`, wdyt?

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                        environment_config   : sdkHarnessImageTag,
+                        environment_type     : 'DOCKER',
+                ]
+        ],
+        [
+                title        : 'ParDo Python Load test: 2GB 100 byte records 200 times',
+                itClass      : 'apache_beam.testing.load_tests.pardo_test:ParDoTest.testParDo',
+                runner       : CommonTestProperties.Runner.PORTABLE,
+                jobProperties: [
+                        job_name             : 'load-tests-python-flink-batch-pardo-2-' + now,
```
Comment (lgajowy) [inline .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: Tiny nit, but I think you can do it even more concise: 

```
"load-tests-python-flink-batch-pardo-2-${now}"
```

wdyt?

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                        number_of_counters   : 1,
+                        parallelism          : 5,
+                        job_endpoint         : 'localhost:8099',
+                        environment_config   : sdkHarnessImageTag,
+                        environment_type     : 'DOCKER',
+                ]
+        ],
+]}
+
+def loadTest = { scope, triggeringContext ->
```
Comment (lgajowy) [inline .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: Comparing to what it looked like, I like how the complexity we used to have in the job file is now gone and the infrastracture providing classes are divided. 👍 Let's work towards making Jenkins jobs even easier to read/implement in the future (I'm open to ideas) :)

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy (hunk 2 above)
Comment (kamilwu) [reply .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: For me, appending string suffixes via '+' operator is a tiny bit more cleaner. I don't think we need to replace this with the ${} syntax.

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy (hunk 1 above)
Comment (kamilwu) [reply .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: +1

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import PhraseTriggeringPostCommitBuilder
+import Flink
+import Docker
+
+String now = new Date().format("MMddHHmmss", TimeZone.getTimeZone('UTC'))
+
+def scenarios = { datasetName, sdkHarnessImageTag -> [
+        [
+                title        : 'ParDo Python Load test: 2GB 100 byte records 10 times',
+                itClass      : 'apache_beam.testing.load_tests.pardo_test:ParDoTest.testParDo',
```
[the remark this answers is not in the document]
Comment (kamilwu) [reply .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: +1 for changing `itClass` to `test`.

> Maybe we can provide a common type (class) for such places

I like this idea. I've already tested such class, but I think I could introduce it in the next PR. Maybe I'll combine it with changes around `LoadTestBuilder` class.

File: .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy (hunk 3 above)
Comment (kamilwu) [reply .test-infra/jenkins/job_LoadTests_ParDo_Flink_Python.groovy]: Great to hear that!

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Python Load Tests ParDo Flink Batch

**Comment by kamilwu** [comment]
@lgajowy I replied to your comments, please take a look once again.

**Comment by iemejia** [comment]
@kamilwu My excuses I somehow totally missed this one, since it seems @lgajowy is quite advanced with the review I won't have much to add. Sorry again.

**Comment by tvalentyn** [comment]
Sorry, I also missed this PR..

**Pull request merged**